wgsl/index.bs

```diff
@@ -4569,8 +4569,12 @@ textureSample(t : texture_depth_cube_array, s : sampler, coords : vec3<f32>, arr
   <tr><td>`offset`<td>
   The optional texel offset applied to the unnormalized texture coordinate
   before sampling the texture. This offset is applied before applying any
-  texture wrapping modes. `offset` values may range from `[-8..7]` in all
-  dimensions. Values outside of this range will be clamped to the nearest limit.
+  texture wrapping modes.<br>
+  `offset` must be compile time constant, and may only be provided as a
+  [module constant](#module-constants), [literal](#literals) or
+  `const_expr` expression (e.g. `vec2<i32>(1, 2)`).<br>
+  Each `offset` component must be at least `-8` and at most `7`. Values outside
```

Comment:
Do we want to be able to use a specialization constant here? If so, we'll have to clamp its value, and the wording wouldn't be correct (i.e. instead of "must be" we should probably say that we are clamping it).

Reply:
Clamping would only be needed in practice if it's a specialization constant. We can specify this *as if* it's always clamped, since the other option would be to fail compilation.
The other option is to aggressively validate it (first in the shader source, then at the pipeline creation like you suggested). We should discuss it on the next call to see which path to follow.

Reply:
Just a heads up that unfortunately I'm unable to attend tonight's call. I'm happy to defer this until next week. Alternatively, I'm also happy for this to land and for us to open another issue about clarifying specializations of offsets.

Reply:
Removed the ability to pass a module constant in, so this is now a non-issue.